cypress/e2e/shared/checks.test.ts

```diff
@@ -737,4 +829,316 @@ describe('Checks', () => {
+  ]
+
+  const thresholdStatuses: CheckStatus[] = [
+    {
+      sourceTimeStamp: calcNanoTimestamp('6m'),
+      level: 'warn',
+      message: 'Stan Goetz - Girl from Imphanema',
+      pair: {key: 'mag', val: 10.3},
+      offset: '-4m',
+      type: 'threshold',
+      name: thresholdCheck.name,
+    },
+    {
+      sourceTimeStamp: calcNanoTimestamp('11m'),
+      level: 'info',
+      message: 'Stanislav Grof - Call of the jaguar',
+      pair: {key: 'mag', val: 1.5},
+      offset: '-9m',
+      type: 'threshold',
+      name: thresholdCheck.name,
+    },
+    {
+      sourceTimeStamp: calcNanoTimestamp('16m'),
+      level: 'crit',
+      message: 'Stanley Kubrick - Barry Lyndon',
+      pair: {key: 'mag', val: 67.2},
+      offset: '-14m',
+      type: 'threshold',
+      name: thresholdCheck.name,
+    },
+    {
+      sourceTimeStamp: calcNanoTimestamp('21m'),
+      level: 'ok',
+      message: 'Nina Simone - Baby just cares for me',
+      pair: {key: 'mag', val: 1.01},
+      offset: '-19m',
+      type: 'threshold',
+      name: thresholdCheck.name,
+    },
+  ]
+
+  const sourceTimeStampSort = (a: CheckStatus, b: CheckStatus) => {
+    if (a.sourceTimeStamp > b.sourceTimeStamp) {
+      return 1
+    } else if (a.sourceTimeStamp < b.sourceTimeStamp) {
+      return -1
+    } else {
+      return 0
+    }
+  }
+
+  const checkStatuses: CheckStatus[] = thresholdStatuses
+    .concat(deadmanStatuses)
+    .sort(sourceTimeStampSort)
+
+  describe('Status history', () => {
+    beforeEach(() => {
+      cy.writeLPDataFromFile({
+        filename: 'data/wumpus01.lp',
+        offset: '20m',
+        stagger: '1m',
+      })
+
+      return cy.get<Organization>('@org').then((org: Organization) => {
+        cy.get<Bucket>('@bucket').then((bucket: Bucket) => {
+          createCheck(deadmanCheck, org, bucket, 'deadmanCheck').then(
+            dmresult => {
+              createCheck(thresholdCheck, org, bucket, 'thresholdCheck').then(
+                thresult => {
+                  cy.log('Now write status messages')
+                  deadmanStatuses.forEach((status: CheckStatus) => {
+                    writeCheckStatusRecord({
+                      check: dmresult,
+                      sourceTimeStamp: status.sourceTimeStamp,
+                      level: status.level,
+                      message: status.message,
+                      pair: status.pair,
+                      offset: status.offset,
+                    })
+                  })
+
+                  thresholdStatuses.forEach((status: CheckStatus) => {
+                    writeCheckStatusRecord({
+                      check: thresult,
+                      sourceTimeStamp: status.sourceTimeStamp,
+                      level: status.level,
+                      message: status.message,
+                      pair: status.pair,
+                      offset: status.offset,
+                    })
+                  })
+                  cy.reload()
```

Comment:
I'm a little nervous about this code block. `writeCheckStatusRecord` performs an async task but we're not waiting on the completion of that work here. I'm afraid that `cy.reload()` will get invoked before all of the `writeCheckStatusRecord` tasks are completed. And `cy.reload()` will terminate any pending API requests. Could we refactor this to add some `.then(() =>` chains?

Reply:
After playing around a bit I discovered that the command `writeLPData()` wraps a 'success' message on a successful return from `/api/v2/write`.  I set the call to `writeCheckStatusRecord()` to then assert that this message is returned.